xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/macros.vm

```diff
@@ -2623,40 +2623,47 @@ Recursive title display detected!##
       <input type="text" id="${escapedId}-filter" class="xwiki-select-filter"
         placeholder="$escapetool.xml($services.localization.render('web.widgets.select.filter.placeholder'))" />
     </p>
-    <div class="xwiki-select-options">
-      <ul>
-        #set ($count = 0)
-        #foreach ($category in $options)
-          <li class="xwiki-select-category">$category.name (<span class="xwiki-select-category-count">$category.options.size()</span>)
-            <ul>
-              #foreach ($option in $category.options)
-                #if ("$!defaultValue" == $$option.value || ($count == 0 && $firstIsDefaultIfDefaultNull && "$!defaultValue" == ''))
-                  #set ($checked = 'checked="checked"')
-                  #set ($class = 'xwiki-select-option-selected')
-                #else
-                  #set ($checked = '')
-                  #set ($class = '')
-                #end
-                #set ($otherData = '')
-                #foreach($data in $option.data.keySet())
-                  #set ($otherData = "$otherData data-${escapetool.xml($data)} = '$escapetool.xml($option.data.get($data))'")
+    <span id="${escapedId}-listbox-label" class="sr-only">
+      $escapetool.xml($services.localization.render('web.widgets.select.listbox.label', [$fieldName]))
+    </span>
+    <span id="${escapedId}-listbox-description" class="sr-only">
+      $escapetool.xml($services.localization.render('web.widgets.select.listbox.description'))
+    </span>
+    <div id="${escapedId}-listbox" role='listbox' class='xwiki-select-options' tabindex='0'
+      aria-labelledby="${escapedId}-listbox-label" aria-describedby="${escapedId}-listbox-description"
+      aria-activedescendant="${fieldName}_0_option">
+      #set ($count = 0)
+      #foreach ($category in $options)
+        <label for="${escapedId}-group${count}">
+          $category.name (<span class="xwiki-select-category-count">$category.options.size()</span>)
```

Comment:
I've just checked and [the HTML standard](https://html.spec.whatwg.org/#the-label-element) says that the `for` attribute must point to an id of a [labelable element](https://html.spec.whatwg.org/#category-label). Unfortunately, elements with role `group` do not seem to be in that category.

Reply:
Addressed in https://github.com/xwiki/xwiki-platform/pull/2182/commits/bcd341777cc54503603a51be015dfd329085ae57

I decided to follow the same structure as the one from the codepen at https://www.w3.org/WAI/ARIA/apg/patterns/listbox/examples/listbox-grouped/ .